src/protocols/secure_channel/PairingSession.cpp

```diff
@@ -60,6 +60,17 @@ void PairingSession::Finish()
 {
     Transport::PeerAddress address = mExchangeCtxt->GetSessionHandle()->AsUnauthenticatedSession()->GetPeerAddress();
 
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    // Release the ExchangeContext that was retained at the beginning of CASE
+    // initiation(both at initiator and responder side). The Retain() was
+    // necessary to prevent the automatic release of the ExchangeContext
+    // after sending the last message of the Exchange and before the call
```

Comment:
Seems like this documentation about why the Retain() is needed should be in the callsites where Retain() is called.

That said.... clearly `mExchangeCtxt` is not usually deleted here.  Why is that different with TCP?  It shouldn't be....

Reply:
Would it be better to fix this by having our member holding the exchange be an ExchangeHandle so we never have dangling pointers as long as we don't null it out?